book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,296 @@
+
+## Constants
+
+### General
+
+- `SLOW_START_INTERVAL: Height`
+- `SLOW_START_SHIFT: Height`
+- `MAX_BLOCK_SUBSIDY: u64`
+- `BLOSSOM_POW_TARGET_SPACING_RATIO: u64`
+- `PRE_BLOSSOM_HALVING_INTERVAL: Height`
+- `POST_BLOSSOM_HALVING_INTERVAL: Height`
+
+### Founders Rewards
+
+- `FOUNDERS_FRACTION_DIVISOR: u64`
+- `FOUNDERS_ADDRESS_COUNT: u32`
+- `FOUNDER_ADDRESSES_MAINNET: [&str; 48]`
+- `FOUNDER_ADDRESSES_TESTNET: [&str; 48]`
+
+### Funding streams
+
+The design suggests to implement the parameters needed for funding streams as:
+
+```
+/// The funding stream receiver categories
+pub enum FundingStreamReceiver {
+    BootstrapProject,
+    ZcashFoundation,
+    MajorGrants,
+}
+
+/// The numerator for each funding stream receiving category 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDING_STREAM_RECEIVER_NUMERATORS: &[(FundingStreamReceiver, u64)] = &[
+    (FundingStreamReceiver::BootstrapProject, 7),
+    (FundingStreamReceiver::ZcashFoundation, 5),
+    (FundingStreamReceiver::MajorGrants, 8),
+];
+
+/// Denominator as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+pub const FUNDING_STREAM_RECEIVER_DENOMINATOR: u64 = 100;
+
+
+/// Start and end Heights for funding streams 
+/// as described in [protocol specification §7.9.1][7.9.1].
+/// 
+/// [7.9.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
+const FUNDING_STREAM_HEIGHT_RANGES: &[(Network, Range<_>)] = &[
+    (Network::Mainnet, Height(1_046_400)..Height(2_726_400)),
+    (Network::Testnet, Height(1_028_500)..Height(2_796_000)),
+];
+```
+
+Each `FundingStreamReceiver` for the mainnet must have 48 payment addresses associated with and 51 for the testnet. 
+
+https://zips.z.cash/zip-0214#mainnet-recipient-addresses
+
+The following constants are needed:
+
+- `FUNDING_STREAM_BP_ADDRESSES_MAINNET: [&str; 48]`
+- `FUNDING_STREAM_ZF_ADDRESSES_MAINNET: [&str; 48]`
+- `FUNDING_STREAM_MG_ADDRESSES_MAINNET: [&str; 48]`
+- `FUNDING_STREAM_BP_ADDRESSES_TESTNET: [&str; 51]`
+- `FUNDING_STREAM_ZF_ADDRESSES_TESTNET: [&str; 51]`
+- `FUNDING_STREAM_MG_ADDRESSES_TESTNET: [&str; 51]`
+
+## General subsidy
+
+The block subsidy and other utility functions are inside the general subsidy category.
+
+https://zips.z.cash/protocol/canopy.pdf#subsidyconcepts
+
+https://zips.z.cash/protocol/canopy.pdf#subsidies
+
+- `block_subsidy(Height, Network) -> Result<Amount<NonNegative>, Error>` - Total block subsidy.
+- `miner_subsidy(Height, Network) -> Result<Amount<NonNegative>, Error>` - Miner portion.
+- `find_output_with_amount(&Transaction, Amount<NonNegative>) -> Vec<transparent::Output>` - Outputs where value equal to Amount.
+- `shielded_coinbase(Height, Network, &Transaction) -> Result<(), Error>` - Validate shielded coinbase rules.
```

Comment:
Why don't these functions need a `&Block` argument?

Reply:
- `block_subsidy(Height, Network)` - fixed amount depending only on height and network
- `miner_subsidy(Height, Network)` - fixed amount depending only on height and network
- `find_output_with_amount(&Transaction, Amount<NonNegative>)` - searches transaction for amount
- `shielded_coinbase(Height, Network, &Transaction)` - validates shielded coinbase rules from `&Transaction`, which does not depend on any other transaction or block info